otter-grader[r] fails with GLIBC linker error

**Describe the bug**
When installing otter-grader with the R extras (e.g., `otter-grader[grading,plugins,r]`) using pip inside a Conda/Mamba environment, the installation fails. The build process fails with a distutils.compilers.C.errors.LinkError, citing undefined references to GLIBC symbols (e.g., powf@GLIBC_2.27, expf@GLIBC_2.27). 

```
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> [48 lines of output]
    /root/miniforge3/envs/otter-env/bin/../lib/gcc/x86_64-conda-linux-gnu/11.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/miniforge3/envs/otter-env/lib/R/lib/../../libblas.so.3: undefined reference to `powf@GLIBC_2.27'
    /root/miniforge3/envs/otter-env/bin/../lib/gcc/x86_64-conda-linux-gnu/11.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/miniforge3/envs/otter-env/lib/R/lib/../../libblas.so.3: undefined reference to `expf@GLIBC_2.27'
    /root/miniforge3/envs/otter-env/bin/../lib/gcc/x86_64-conda-linux-gnu/11.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/miniforge3/envs/otter-env/lib/R/lib/../../libblas.so.3: undefined reference to `logf@GLIBC_2.27'
    collect2: error: ld returned 1 exit status
    ...
    distutils.compilers.C.errors.LinkError: command '/root/miniforge3/envs/otter-env/bin/x86_64-conda-linux-gnu-cc' failed with exit code 1
    [end of output]
```

**To Reproduce**
I have attached the ZIP file from `otter generate` when uploading to Gradescope (tests are removed for sake of not revealing private solutions). 

[Archive.zip](https://github.com/user-attachments/files/22436647/Archive.zip)

**Expected behavior**
otter-grader should build successfully. 

**Versions**
Python version: 3.10.18
Otter-Grader version: 6.1.5

Additional package versions:
    dill: 0.4.0
    fica: 0.4.1
    IPython: 8.37.0
    nbconvert: 7.16.6
    nbformat: 5.10.4

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+**v6.1.6:**
+
+* Fix R grading image builds by updating grading image to install rpy2 from conda-forge instead of using the `otter-grader[r]` extra per [#947](https://github.com/ucbds-infra/otter-grader/issues/947)
+
 **v6.1.5:**
 
 * Pin miniforge version in grading image to v25.3.1-0

--- a/otter/generate/__init__.py
+++ b/otter/generate/__init__.py
@@ -79,18 +79,15 @@ class CondaEnvironment:
                     "r-testthat",
                     "r-usethis",
                     f"r-ottr=={OTTR_VERSION}",
+                    "rpy2",
                 ]
             )
-
-        r_extra = ""
-        if self.is_r:
-            r_extra = ",r"
 
         pip_deps = (
             self.requirements
             if self.overwrite_requirements
             else [
-                f"otter-grader[grading,plugins{r_extra}]=={__version__}",
+                f"otter-grader[grading,plugins]=={__version__}",
                 *self.requirements,
             ]
         )

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,6 +55,8 @@ wrapt = "^1.16.0"
 [tool.poetry.extras]
 grading = ["ipykernel", "jupyter_client", "pypdf"]
 plugins = ["google-api-python-client", "google-auth-oauthlib", "gspread", "six"]
+# IMPORTANT: The r extra is not used to construct the grading environment, so any new dependencies
+# for R grading  also need to be added to the generated R environment.yml file in otter generate.
 r = ["rpy2"]
 
 [tool.poetry.group.dev]

--- a/test/test_assign/files/rmd-autograder-correct/environment.yml
+++ b/test/test_assign/files/rmd-autograder-correct/environment.yml
@@ -24,5 +24,6 @@ dependencies:
   - r-testthat
   - r-usethis
   - r-ottr==1.5.2
+  - rpy2
   - pip:
-      - otter-grader[grading,plugins,r]==6.1.5
+      - otter-grader[grading,plugins]==6.1.5

--- a/test/test_generate/files/autograder-r-correct/environment.yml
+++ b/test/test_generate/files/autograder-r-correct/environment.yml
@@ -24,5 +24,6 @@ dependencies:
   - r-testthat
   - r-usethis
   - r-ottr==1.5.2
+  - rpy2
   - pip:
-      - otter-grader[grading,plugins,r]==6.1.5
+      - otter-grader[grading,plugins]==6.1.5

--- a/test/test_generate/files/autograder-r-requirements-correct/environment.yml
+++ b/test/test_generate/files/autograder-r-requirements-correct/environment.yml
@@ -24,5 +24,6 @@ dependencies:
   - r-testthat
   - r-usethis
   - r-ottr==1.5.2
+  - rpy2
   - pip:
-      - otter-grader[grading,plugins,r]==6.1.5
+      - otter-grader[grading,plugins]==6.1.5

